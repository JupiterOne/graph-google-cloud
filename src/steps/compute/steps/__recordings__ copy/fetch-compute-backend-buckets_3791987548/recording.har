{
  "log": {
    "_recordingName": "fetch-compute-backend-buckets",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 709,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.16.0"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "709"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1225,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1225,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 24 Jan 2024 12:06:55 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-24T12:06:54.833Z",
        "time": 131,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 131
        }
      },
      {
        "_id": "fdc2c329adcfe7366047fe0680fdf7d5",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.16.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "compute.googleapis.com"
            }
          ],
          "headersSize": 1326,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://compute.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/backendBuckets"
        },
        "response": {
          "bodySize": 576,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 576,
            "text": "{\"kind\":\"compute#backendBucketList\",\"id\":\"projects/j1-gc-integration-dev-v3/global/backendBuckets\",\"items\":[{\"kind\":\"compute#backendBucket\",\"id\":\"5659178066310414864\",\"creationTimestamp\":\"2024-01-24T04:05:19.115-08:00\",\"name\":\"delete\",\"description\":\"\",\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/backendBuckets/delete\",\"bucketName\":\"gaston-delete\",\"enableCdn\":true,\"cdnPolicy\":{\"requestCoalescing\":true,\"cacheMode\":\"CACHE_ALL_STATIC\",\"defaultTtl\":3600,\"maxTtl\":86400,\"clientTtl\":3600,\"negativeCaching\":false,\"serveWhileStale\":0},\"compressionMode\":\"DISABLED\"}],\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/backendBuckets\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "uSQesqzvJcgFo3GMvTNi2IcffxY=/fLgkUMqHhFTfq8OyQ5Fv7QqRopU="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 24 Jan 2024 12:06:55 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 432,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-24T12:06:54.972Z",
        "time": 395,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 395
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
