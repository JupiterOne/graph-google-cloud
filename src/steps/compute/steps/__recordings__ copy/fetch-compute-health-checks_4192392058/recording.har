{
  "log": {
    "_recordingName": "fetch-compute-health-checks",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 709,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.16.0"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "709"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1239,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1239,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 24 Jan 2024 12:24:18 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-24T12:24:18.399Z",
        "time": 161,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 161
        }
      },
      {
        "_id": "38f64d538b96b56a8abc287c9767d751",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.16.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "compute.googleapis.com"
            }
          ],
          "headersSize": 1324,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://compute.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/healthChecks"
        },
        "response": {
          "bodySize": 509,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 509,
            "text": "{\"kind\":\"compute#healthCheckList\",\"id\":\"projects/j1-gc-integration-dev-v3/global/healthChecks\",\"items\":[{\"kind\":\"compute#healthCheck\",\"id\":\"819908078879682790\",\"creationTimestamp\":\"2021-05-31T09:25:45.951-07:00\",\"name\":\"load-balancer-health-check\",\"checkIntervalSec\":1,\"timeoutSec\":1,\"unhealthyThreshold\":2,\"healthyThreshold\":2,\"type\":\"TCP\",\"tcpHealthCheck\":{\"port\":80,\"proxyHeader\":\"NONE\"},\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/healthChecks/load-balancer-health-check\"}],\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/healthChecks\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "HVK7Q5sll3wC361kIUFbw4H1FWM=/j5BRt-2MJMvSW7SG0k_crZ-ZAtE="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 24 Jan 2024 12:24:19 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 432,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-24T12:24:18.569Z",
        "time": 426,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 426
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
