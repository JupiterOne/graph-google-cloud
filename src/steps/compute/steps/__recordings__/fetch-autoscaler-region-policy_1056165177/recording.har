{
  "log": {
    "_recordingName": "fetch-autoscaler-region-policy",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 727,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "727"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1174,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1174,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 21 May 2024 11:34:35 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-21T11:34:35.391Z",
        "time": 286,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 286
        }
      },
      {
        "_id": "fc6c41a4fc74b579d77479a836fdba46",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "compute.googleapis.com"
            }
          ],
          "headersSize": 1333,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://compute.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-east1/autoscalers"
        },
        "response": {
          "bodySize": 331,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 331,
            "text": "{\"kind\":\"compute#regionAutoscalerList\",\"id\":\"projects/j1-gc-integration-dev-v3/regions/us-east1/autoscalers\",\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-east1/autoscalers\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "S10ca0l_xaj2sr6KSBUmrgpncl4=/SlIJogddj45G6V2WfuqyXrYI3dc="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 21 May 2024 11:34:38 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 432,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-21T11:34:35.696Z",
        "time": 2780,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 2780
        }
      },
      {
        "_id": "d6b39180ca79e1b4b4714266ac2e4d3f",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "compute.googleapis.com"
            }
          ],
          "headersSize": 1333,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://compute.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-east4/autoscalers"
        },
        "response": {
          "bodySize": 287,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 287,
            "text": "{\"kind\":\"compute#regionAutoscalerList\",\"id\":\"projects/j1-gc-integration-dev-v3/regions/us-east4/autoscalers\",\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-east4/autoscalers\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "q_doIrgIviIhkiutFfsJrgGNGVE=/KHCamfPMc4M-qcjHgph34aqy1o4="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 21 May 2024 11:34:39 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 432,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-21T11:34:38.486Z",
        "time": 1550,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1550
        }
      },
      {
        "_id": "00dcdc9b7a8b6700431745e4edc4c41e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "compute.googleapis.com"
            }
          ],
          "headersSize": 1336,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://compute.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-central1/autoscalers"
        },
        "response": {
          "bodySize": 889,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 889,
            "text": "{\"kind\":\"compute#regionAutoscalerList\",\"id\":\"projects/j1-gc-integration-dev-v3/regions/us-central1/autoscalers\",\"items\":[{\"kind\":\"compute#autoscaler\",\"id\":\"4850256074544420443\",\"creationTimestamp\":\"2024-05-20T03:40:52.802-07:00\",\"name\":\"rbs-rigm\",\"target\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-central1/instanceGroupManagers/rbs-rigm\",\"autoscalingPolicy\":{\"minNumReplicas\":1,\"maxNumReplicas\":6,\"coolDownPeriodSec\":60,\"loadBalancingUtilization\":{\"utilizationTarget\":0.3},\"mode\":\"ON\"},\"region\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-central1\",\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-central1/autoscalers/rbs-rigm\",\"status\":\"ACTIVE\",\"statusDetails\":[{\"message\":\"The autoscaler is configured to scale based on a load balancing signal but the instance group has not received any utilization data from the load balancer or the utilization is constantly 0. Check that the load balancing configuration is working.\",\"type\":\"MISSING_LOAD_BALANCING_DATA_POINTS\"}],\"recommendedSize\":1}],\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/regions/us-central1/autoscalers\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "l1FVhQnPqkPMaaR4_2_vwLXjBFg=/0nEnbveX4uwtWz3t8qYxwhli6sk="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 21 May 2024 11:34:41 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 432,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-21T11:34:40.046Z",
        "time": 1589,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1589
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
