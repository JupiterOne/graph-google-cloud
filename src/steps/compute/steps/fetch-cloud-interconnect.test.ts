import {
  Recording,
  createMockStepExecutionContext,
} from '@jupiterone/integration-sdk-testing';
import {
  ENTITY_CLASS_CLOUD_INTERCONNECT,
  ENTITY_TYPE_CLOUD_INTERCONNECT,
} from '../constants';
import { setupGoogleCloudRecording } from '../../../../test/recording';
import { integrationConfig } from '../../../../test/config';
import { IntegrationConfig } from '../../..';
import { fetchCloudInterconnect } from './fetch-cloud-interconnect';

describe('#fetchCloudInterconnect', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchCloudInterconnect',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: {
        ...integrationConfig,
        serviceAccountKeyFile: integrationConfig.serviceAccountKeyFile.replace(
          'j1-gc-integration-dev-v2',
          'j1-gc-integration-dev-v3',
        ),
        serviceAccountKeyConfig: {
          ...integrationConfig.serviceAccountKeyConfig,
          project_id: 'j1-gc-integration-dev-v3',
        },
      },
    });

    await fetchCloudInterconnect(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === ENTITY_TYPE_CLOUD_INTERCONNECT,
      ),
    ).toMatchGraphObjectSchema({
      _class: ENTITY_CLASS_CLOUD_INTERCONNECT,
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: ENTITY_TYPE_CLOUD_INTERCONNECT },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          CIDR: { exclude: true },
          internal: { exclude: true },
          id: { type: 'string' },
          name: { type: 'string' },
          location: { type: 'string' },
          interConnectType: { type: 'string' },
          kind: { type: 'string' },
          public: { type: 'boolean' },
          adminEnabled: { type: 'boolean' },
          createdOn: { type: 'number' },
        },
      },
    });
  });
});
