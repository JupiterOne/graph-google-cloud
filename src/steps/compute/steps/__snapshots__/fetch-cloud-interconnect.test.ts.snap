// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchCloudInterconnect should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Network",
      ],
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnects/demo-interconnect",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "adminEnabled": true,
            "circuitInfos": [
              {
                "customerDemarcId": "",
                "googleCircuitId": "GCP-00008",
                "googleDemarcId": "VOCUS:MMR-A: Ground 7A, Frontcomms, Cab FC4, PP-HAAA12, RU21-24 FP3 Fibers 63,64 - LC",
              },
            ],
            "creationTimestamp": "2024-04-22T23:53:41.482-07:00",
            "customerName": "demo-customer",
            "description": "Example Compute Cloud Interconnect",
            "googleReferenceId": "1134185888",
            "id": "3602008363502330906",
            "interconnectType": "IT_PRIVATE",
            "kind": "compute#interconnect",
            "labelFingerprint": "42WmSpB8rSM=",
            "linkType": "LINK_TYPE_ETHERNET_10G_LR",
            "location": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnectLocations/akl-zone1-1353",
            "macsecEnabled": false,
            "name": "demo-interconnect",
            "operationalStatus": "OS_UNPROVISIONED",
            "provisionedLinkCount": 1,
            "requestedLinkCount": 1,
            "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnects/demo-interconnect",
            "state": "UNPROVISIONED",
          },
        },
      ],
      "_type": "google_cloud_interconnect",
      "adminEnabled": true,
      "createdOn": 1713855221482,
      "description": "Example Compute Cloud Interconnect",
      "displayName": "demo-interconnect",
      "id": "3602008363502330906",
      "interConnectType": "IT_PRIVATE",
      "kind": "compute#interconnect",
      "location": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v3/global/interconnectLocations/akl-zone1-1353",
      "name": "demo-interconnect",
      "public": true,
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "google_cloud_interconnect",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
